proposals/generic-ref-struct.md

```diff
@@ -0,0 +1,424 @@
+Unconstraining our generics
+=====
+
+## Summary
+This proposal is an aggregation of several different proposals for making `ref struct` more generally usable in our type system. The goal being to unify code paths today that have to specialize on `ref struct` and other types do to the inability of `ref struct` to participate in generics and implement interfaces. This proposal will allow us to generalize on `ref struct` as naturally as other types. 
+
+## Motivation
+This feature will allow developers to remove `ref struct` specific code by instead letting them implement interfaces that are used in generic contexts to generalize algorithms. For example an explicit goal is to allow `ref struct` to implement `ISpanFormattable`. The `ISpanFormattable` interface and `ref struct` are used in low level perf sensitive areas but our current language limitations prevent them from being used together.
+
+This will also allow for combinations like `Span<Span<T>>`. This is a long desire of the runtime team who often want to represent collections of `Span<T>` but are unable to do so due to our current limitations.
```

Comment:
```suggestion
This will also allow for combinations like `Span<Span<T>>`. This has long been a desire of the runtime team, who often want to represent collections of `Span<T>` but are unable to do so due to our current limitations.
```

Reply:
Or alternately
```suggestion
This will also allow for combinations like `Span<Span<T>>`. This is a long-held desire of the runtime team who often want to represent collections of `Span<T>` but are unable to do so due to our current limitations.
```